Public/Src/Cache/ContentStore/Distributed/NuCache/EventStreaming/EventHubContentLocationEventStore.cs

```diff
@@ -112,30 +112,52 @@ protected override async Task<BoolResult> StartupCoreAsync(OperationContext cont
                 return baseInitializeResult;
             }
 
-            var connectionStringBuilder =
-                new EventHubsConnectionStringBuilder(_configuration.EventHubConnectionString)
-                {
-                    EntityPath = _configuration.EventHubName,
-                };
-
             _currentEventProcessor = new Processor(context, this);
 
-            // Retry behavior in the Azure Event Hubs Client Library is controlled by the RetryPolicy property on the EventHubClient class.
-            // The default policy retries with exponential backoff when Azure Event Hub returns a transient EventHubsException or an OperationCanceledException.
-            _eventHubClient = EventHubClient.CreateFromConnectionString(connectionStringBuilder.ToString());
-            _partitionSender = _eventHubClient.CreatePartitionSender(PartitionId);
+            RegenerateEventHubClientFromCredentials(new EventHubsConnectionStringBuilder(_configuration.EventHubConnectionString)
+            {
+                EntityPath = _configuration.EventHubName,
+            });
 
             return BoolResult.Success;
         }
 
+        private readonly object _eventHubLock = new object();
+
+        /// <nodoc />
+        public void UpdateCredentials(OperationContext context, EventHubsConnectionStringBuilder connectionStringBuilder)
+        {
+            lock (_eventHubLock)
+            {
+                DoSuspendProcessing(context).ThrowIfFailure();
+                RegenerateEventHubClientFromCredentials(connectionStringBuilder);
+                DoStartProcessing(context, GetLastProcessedSequencePoint()).ThrowIfFailure();
+            }
+        }
+
+        private void RegenerateEventHubClientFromCredentials(EventHubsConnectionStringBuilder connectionStringBuilder)

```

Comment:
I would say the name should be: CreateEventHubClient.

We usually creating instances, not "regenerating" them, IMO.